app/Http/Controllers/API/Posts/PostsController.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace Ushahidi\App\Http\Controllers\API\Posts;
+
+use Ushahidi\App\Http\Controllers\RESTController;
+use Illuminate\Http\Request;
+use Ushahidi\Core\Usecase;
+
+/**
+ * Ushahidi API Posts Controller
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application\Controllers
+ * @copyright  2013 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+class PostsController extends RESTController
+{
+	/**
+	 * @var int Post Parent ID
+	 */
+	protected $parentId = null;
+
+	/**
+	 * @var string Post Type
+	 */
+	protected $postType = 'report';
+
+	// Ushahidi_Rest
+	protected function getResource()
+	{
+		return 'posts';
+	}
+
+	protected function getIdentifiers(Request $request)
+	{
+		return $this->getRouteParams($request) + [
+			'type'      => $this->postType
+		];
+	}
+
+	protected function getFilters(Request $request)
+	{
+        $params = $this->getRouteParams($request);
+		return $request->query() + [
+			'type'      => $this->postType,
+			'parent'    => isset($params['parent_id']) ? $params['parent_id'] : null,
+		];
+	}
+
+	protected function getPayload(Request $request)
```

Comment:
No. Filters are built from the query string, payload is built from the JSON body of a request.